compiler/modules/types/int.bal

```diff
@@ -133,6 +133,21 @@ function intSubtypeComplement(SubtypeData d) returns SubtypeData {
     return rangeListComplement(v).cloneReadOnly();
 }
 
+// `min` and `max` are inclusive
+function intSubtypeOverlapRange(IntSubtype subtype, int min, int max) returns boolean {
+    IntSubtype range = [{ min, max }];
+    return intSubtypeIntersect(subtype, range) != false;
+}
+
+function intSubtypeContainsGreaterThan(IntSubtype subtype, int value) returns boolean {
```

Comment:
Can we instead have:

```
function intSubtypeMax(IntSubtype subtype) returns int {
    return subtype[subtype.length() - 1].max;
}
```

Similarly for intSubtypeMin. Then use in intConstraints function in core.